vocabularies/Analytics.json

```diff
@@ -161,6 +161,102 @@
+          },
+          {
+            "$Name": "ExpandLevel",
+            "$Type": "Edm.Int64",
+            "@Core.Description": "Number of levels to be expanded, counting from left",
+            "@Core.LongDescription": "The function result comprises the leveled hierarchy with the expanded levels and the following level as leaves in preorder."
+          },
+          {
+            "$Name": "Aggregation",
+            "$Collection": true,
+            "@Core.Description": "Properties to aggregate for all result entries on all levels",
+            "@Core.LongDescription": "All properties in this collection must be custom aggregates."
+          },
+          {
+            "$Name": "SiblingOrder",
+            "$Collection": true,
+            "$Type": "Analytics.AutoExpandSiblingOrder",
+            "@Core.Description": "Sort specification to apply to all direct descendants of a given entry in the resulting leveled hierarchy"
+          },
+          {
+            "$Name": "BeforeAggregationFilter",
+            "@Core.Description": "Expression valid for `filter` transformation to restrict the input set before any further procressing",
+            "@Core.OptionalParameter": {}
+          },
+          {
+            "$Name": "AggregatedValuesLeafFilter",
+            "@Core.Description": "Expression valid for `filter` transformation to restrict the input set at the most detailed grouping level with conditions on aggregated values",
+            "@Core.OptionalParameter": {}
+          },
+          {
+            "$Name": "Skip",
+            "$Type": "Edm.Int64",
+            "@Core.Description": "Number of entries to skip from the top of the fully ordered result",
+            "@Core.OptionalParameter": { "DefaultValue": "0" }
+          },
+          {
+            "$Name": "Top",
+            "$Type": "Edm.Int64",
+            "@Core.Description": "Number of entries to return from the result set after any skipping (absent means all)",
+            "@Core.OptionalParameter": {}
+          },
+          {
+            "$Name": "ResultEntriesCount",
+            "$Type": "Edm.Bool",
+            "@Core.Description": "Whether to return the total number of entries in the result independent of Skip/Top",
+            "@Core.OptionalParameter": { "DefaultValue": "false" }
+          }
+        ],
+        "$ReturnType": { "$Collection": true, "$Type": "Edm.EntityType" }
+      }
+    ],
+    "AutoExpandLevel": {
+      "$Kind": "ComplexType",
+      "@Common.Experimental": true,
+      "P": { "$Collection": true, "@Core.Description": "A non-empty set of property names constituting a level" }
```

Comment:
Does the description fit? We would need the the information of `group` and `groupLevels` as described in [sap.ui.model.odata.v4.ODataListBinding#setAggregation](https://ui5.sap.com/#/api/sap.ui.model.odata.v4.ODataListBinding%23methods/setAggregation) with `group` denoting all properties used for grouping the result data on leaf level and `groupLevels` denoting the properties used for visual grouping.

Reply:
`Levels` is a collection of collections (technically, a collection of records with one collection-valued property). The set of grouping properties for the leaf level is the union of all nested collections, and the set of properties for visual grouping is the first nested collection. But you also need the grouping properties for the intermediate levels, which shall be auto-expanded.

Reply:
So I have something like `[['Country','Region'],['Segment']]` meaning that the leaf level is grouped by `Country`, `Region` and `Segment` and that there are two levels in the visual grouping, first the `Country` and then the `Region`. Is that understanding correct?

Reply:
No. Leaves are grouped by `Country`, `Region` and `Segment`. There is one more level grouped by `Country` and `Region` (this is called "visual grouping", right?).

Reply:
To achieve what I want, I would need `[['Country'],['Region'],['Segment']]`?
The flexibility that [sap.ui.model.odata.v4.ODataListBinding#setAggregation](https://ui5.sap.com/#/api/sap.ui.model.odata.v4.ODataListBinding%23methods/setAggregation) provides with `group` and `groupLevels` is required for this function as well. I.e. we need to be able to specify an arbitrary number of groupable properties that are used for visual grouping. And the number of (properties in) `groupLevels` determines the number of visual groups. And then there can be an arbitrary number of additional properties in `group` that would be used additionally for the leaf level.